### ISSUE TITLE
Introduce `StructImplementation` and associated methods

### ISSUE THREAD

**Issue by Thunkar** [noir-lang/noir#5912]
### Problem

Currently there's only `StructDefinition`. Sometimes it is useful to manipulate a struct's impl, much like `TraitImplementation` (mostly to add methods).

### Happy Case

Something like this can be done:

```noir
comptime fn get_imp(s: StructDefinition) {
  let implementation = s.get_impl();
  let methods = implementation.methods();
  let new_method = ...;
  implementation.set_methods(methods.push_back(new_method));
}
```

### Workaround

None

### Project Impact

Blocker

### Would you like to submit a PR for this Issue?

None

**Comment by jfecher** [comment]
> Sometimes it is useful to manipulate a struct's impl, much like TraitImplementation (mostly to add methods).

I think this use case may be covered by allowing unquoting impls at top-level once [foo](https://github.com/noir-lang/noir/pull/5927) is merged:

```rs
comptime fn get_imp(s: StructDefinition) -> Quoted {
  let t = s.as_type();
  let generics = s.generics()
    .map(|g| quote { $g })
    .join(quote {,});

  quote {
      impl<$generics> $t {
          pub fn my_new_method(self) { ... }
      }
  }
}
```

**Comment by asterite** [comment]
Fixed by #5927
